junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/MethodBasedTestDescriptor.java

```diff
@@ -97,7 +98,11 @@ public ExclusiveResourceCollector getExclusiveResourceCollector() {
 
 	@Override
 	public Set<ResourceLocksProvider.Lock> evaluateResourceLocksProvider(ResourceLocksProvider provider) {
-		return provider.provideForMethod(getTestClass(), getTestMethod());
+		List<Class<?>> enclosingInstanceTypes = getParent() //
+				.filter(ClassBasedTestDescriptor.class::isInstance) //
+				.map(parent -> ((ClassBasedTestDescriptor) parent).getEnclosingTestClasses()) //
+				.orElseGet(Collections::emptyList);
```

Comment:
I haven't tested this in an IDE, but I often prefer the following approach, even if it requires an additional `map()` invocation.

```suggestion
				.filter(ClassBasedTestDescriptor.class::isInstance) //
				.map(ClassBasedTestDescriptor.class::cast) //
				.map(ClassBasedTestDescriptor::getEnclosingTestClasses) //
				.orElseGet(Collections::emptyList);
```